sdk/servicebus/azure-servicebus/azure/servicebus/aio/_servicebus_sender_async.py

```diff
@@ -278,6 +279,12 @@ async def send_messages(self, message):
             pass
         if isinstance(message, BatchMessage) and len(message) == 0:  # pylint: disable=len-as-condition
             raise ValueError("A BatchMessage or list of Message must have at least one Message")
+        if not isinstance(message, BatchMessage):
+            if not isinstance(message, Message):
+                raise TypeError("Can only send azure.servicebus.<BatchMessage,Message> or lists of Messages.")
+            # This allows resending an already sent message; not needed for batch.  Perhaps should be in uamqp?
+            message.message.state = MessageState.WaitingToBeSent
+            message.message._response = None # pylint: disable=protected-access
```

Comment:
I'm wondering if this part could be moved into `copy_messages_to_sendable_if_needed` so that we could maintain the logic of resendable message more compact and easy to maintain.

And we could probably also rename the method `copy_messages_to_sendable_if_needed` to something like "transfer_to_resendable_message"?

Reply:
I considered upleveling that, didn't because "it'll be overkill to recreate" but we can potentially be smarter than that.  I'll see what can be done.

I'm less sold on the renaming, since transform is less explicit on the potential create aspect, but I get the goal you're going for, so will brainstorm.

Reply:
Spent far too long bashing my head against that name, couldn't find anything better than your transform example that didn't lose precision or clarity.  Works for me.